src/configFile.ts

```diff
@@ -0,0 +1,73 @@
+import { parse } from "toml";
+import { readFile } from "fs";
+import findUp from "find-up";
+interface foremanConfig {
+  tools: {
+    [tool_name: string]: foremanTool
+  };
+}
+
+interface foremanTool {
+  source?: string;
+  github?: string;
+  gitlab?: string;
+  version: string;
+}
+
+const MANIFEST = "foreman.toml";
+
+function checkSameOrgToolSpecs(manifestContent: foremanConfig): boolean {
+  const tools = manifestContent.tools;
+  if (tools == null) {
+    throw new Error("Tools section in Foreman config not found");
+  }
+
+  const orgs: string[] = [];
+  for (const tool_name in tools) {
+    const tool_spec = tools[tool_name];
+    let source = tool_spec["source"];
+    if (source == null) {
+      source = tool_spec["github"];
+    }
+    if (source == null) {
+      continue;
+    }
+
+    const source_array = source.split("/");
+    const org = source_array[0];
+
+    if (org == null) {
+      throw new Error(
+        `Org not found in tool spec definition for: ${tool_name}`
+      );
+    }
+    orgs.push(org.toLowerCase());
+  }
+  if (orgs.length == 0) {
+    return true;
+  }
+  return orgs.every(val => val === orgs[0]);
+}
+
+async function checkSameOrgInConfig(): Promise<void> {
+  const manifestPath = await findUp(MANIFEST);
+  if (manifestPath == null) {
+    throw new Error("Foreman config file could not be found");
+  }
+
+  await readFile(manifestPath, "utf8", (err, data) => {
+    if (err) {
+      throw new Error("Could not read Foreman config file");
```

Comment:
I wonder if we should make it clear that `setup-foreman` is encountering this error, not `foreman` itself. Just in case we get out of sync somehow, or they're using a version of foreman that doesn't line up with this, we should clarify that the action hit a snag and not the CLI itself.